### ISSUE TITLE
Replacing the usage of ObjectManager in Magento_Catalog Controllers

### ISSUE THREAD

**Issue by drpayyne** [magento/magento2#24646]
### Summary
The usage of ObjectManager is discouraged and this issue is to address the extremely high and unnecessary usage of ObjectManagers in core code. This prevents type hinting as ObjectManagers can return any class. It also scatters the dependencies across the class hindering clear understanding of a class' dependencies. Also, ObjectManagers are used where Factories and Constructor Dependency Injections are preferred.

### Examples

- `$resultPageFactory = $this->_objectManager->get(\Magento\Framework\View\Result\PageFactory::class);`
- `$category = $this->_objectManager->create(\Magento\Catalog\Model\Category::class)->load($categoryId);`
- And more...

### Proposed solution
Refactoring ObjectManagers to Constructor Dependency Injections or Factories where necessary and possible.

### Steps To Reproduce
- Use `2.3-develop` branch
- Navigate to `app/code/Magento/Catalog/Controller/Adminhtml/Category/Add` to find usages of ObjectManager in the code
- Similarly, ObjectManagers are found in multiple classes unnecessarily 

### Expected
```
public function __construct (
    ...
    \Magento\Framework\View\Result\PageFactory $pageFactory
) {
    ....
    $this->pageFactory = $pageFactory;
}
```

### Actual
```
$resultPageFactory = $this->_objectManager->get(\Magento\Framework\View\Result\PageFactory::class);
```

**Comment by engcom-Bravo** [comment]
Hello @sdzhepa.
This issue seems to be a feature request.
Will you please transfer it to the feature requests repository. Thank You.

**Comment by drpayyne** [comment]
@sdzhepa May I work on this issue, please? Also, isn't this a technical issue rather than a feature request as labelled by @engcom-Bravo?

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sudheers-kensium 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-20325` were created

**Issue Available**: @sudheers-kensium, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @drpayyne. Thank you for your report. 
The issue has been fixed in magento/magento2#24661 by @drpayyne in 2.3-develop branch
Related commit(s):
 - [2b5b739506f34cc13dbbe91b50ff9fb6ceadfe6d](https://github.com/magento/magento2/commit/2b5b739506f34cc13dbbe91b50ff9fb6ceadfe6d)

The fix will be available with the upcoming 2.3.4 release.
